dbms/src/Storages/DeltaMerge/File/ColumnStream.cpp

```diff
@@ -94,17 +94,15 @@ class MarkLoader
     }
     MarksInCompressedFilePtr loadColMarkFromMetav2To(const MarksInCompressedFilePtr & res, size_t bytes_size)
     {
-        auto info = reader.dmfile->merged_sub_file_infos.find(reader.dmfile->colMarkFileName(file_name_base));
-        if (info == reader.dmfile->merged_sub_file_infos.end())
+        const auto * dmfile_meta = typeid_cast<const DMFileMetaV2 *>(reader.dmfile->meta.get());
```

Comment:
Add `assert(dmfile_meta != nullptr)` check

Reply:
Assert is too dangerous. It is as dangerous as not using assert at all, because nullptr always result in a crash, just like assert. How about using a RUNTIME_CHECK?

Reply:
`loadColMarkFromMetav2To` is a special dispatch for "metav2", so I'm assuming the caller ensure that these code is only called under "metav2" code path in runtime. A `assert` here in debug mode should be ok.

Reply:
Since we only cast when we make sure it is metav2, so `dmfile_meta` should not be `nullptr`.  I think `assert` is enough